### PR TITLE
JWT refactor and cleanup, move components to separate module and files

### PR DESCRIPTION
**Pull request by spetz**

### CODE REVIEW

File: server/src/http/jwt/middleware.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+fn should_skip_auth(path: &str) -> bool {
+    matches!(
+        path,
+        "/" | "/metrics"
+            | "/ping"
+            | "/users/login"
+            | "/users/refresh-token"
+            | "/personal-access-tokens/login"
```
Comment (hubcio) [inline server/src/http/jwt/middleware.rs]: move this to consts, eg. 
```rust
const UNAUTHORIZED_PATHS: &[&str] = &[
    "/",
    "/metrics",
    "/ping",
    "/users/login",
    "/users/refresh-token",
    "/personal-access-tokens/login",
];
```

and use something like
```rust
fn should_skip_auth(path: &str) -> bool {
    UNAUTHORIZED_PATHS.contains(&path)
}
```

File: server/src/http/jwt/middleware.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    middleware::Next,
+    response::Response,
+};
+use std::sync::Arc;
+
+const AUTHORIZATION: &str = "authorization";
+const BEARER: &str = "Bearer ";
+const UNAUTHORIZED: StatusCode = StatusCode::UNAUTHORIZED;
+
+pub async fn jwt_auth<T>(
```
Comment (hubcio) [inline server/src/http/jwt/middleware.rs:24]: why not
```rust

pub async fn jwt_auth(
    State(state): State<Arc<AppState>>,
    mut request: Request<_>,
    next: Next<_>,
)
```
isn't type inferred from `Request<T>`?

File: server/src/http/jwt/storage.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                Err(err) => {
+                    error!("Cannot load refresh token. Error: {}", err);
+                    return Err(Error::CannotLoadResource(key.to_string()));
+                }
+            };
+            refresh_tokens.push(token);
+        }
+        info!("Loaded {} refresh tokens", refresh_tokens.len());
+        Ok(refresh_tokens)
+    }
```
Comment (hubcio) [inline server/src/http/jwt/storage.rs:74]: refactor it like so, it`ll be more idiomatic:

```rust
    pub fn load_all_refresh_tokens(&self) -> Result<Vec<RefreshToken>, Error> {
        let key = format!("{REFRESH_TOKENS_KEY_PREFIX}:");

        let refresh_tokens: Result<Vec<_>, _> = self
            .db
            .scan_prefix(&key)
            .map(|data| {
                let (hash, value) = data.map_err(|err| {
                    error!("Cannot load refresh token. Error: {}", err);
                    Error::CannotLoadResource(key.clone())
                })?;

                let mut token = rmp_serde::from_slice::<RefreshToken>(&value).map_err(|err| {
                    error!("Cannot deserialize refresh token. Error: {}", err);
                    Error::CannotDeserializeResource(key.clone())
                })?;

                token.token_hash = String::from_utf8(hash).map_err(|_| {
                    error!("Cannot convert hash to UTF-8 string");
                    Error::SOMEERROR(key.clone())
                })?;

                Ok(token)
            })
            .collect();

        let refresh_tokens = refresh_tokens?;
        info!("Loaded {} refresh tokens", refresh_tokens.len());
        Ok(refresh_tokens)
    }
    
```

File: server/src/http/jwt/middleware.rs (hunk 1 above)
Comment (spetz) [reply server/src/http/jwt/middleware.rs]: Done

File: server/src/http/jwt/middleware.rs (hunk 2 above)
Comment (spetz) [reply server/src/http/jwt/middleware.rs:24]: It won't work/

**Review by hubcio — APPROVED** [review]

**Pull request merged**